platform-service/modules/resource-group-service/src/main/java/io/harness/resourcegroup/framework/v2/remote/resource/HarnessResourceGroupResourceImpl.java

```
@@ -102,7 +111,21 @@ public ResponseDTO<PageResponse<ResourceGroupResponse>> list(@AccountIdentifier
 
   public ResponseDTO<PageResponse<ResourceGroupResponse>> list(
       ResourceGroupFilterDTO resourceGroupFilterDTO, String accountIdentifier, PageRequest pageRequest) {
-    return ResponseDTO.newResponse(getNGPageResponse(resourceGroupService.list(resourceGroupFilterDTO, pageRequest)));
+    if (accessControlClient.hasAccess(
+            ResourceScope.of(resourceGroupFilterDTO.getAccountIdentifier(), resourceGroupFilterDTO.getOrgIdentifier(),
+                resourceGroupFilterDTO.getProjectIdentifier()),
+            Resource.of(RESOURCE_GROUP, null), VIEW_RESOURCEGROUP_PERMISSION)) {
+      return ResponseDTO.newResponse(getNGPageResponse(resourceGroupService.list(resourceGroupFilterDTO, pageRequest)));
+    }
+
+    Page<ResourceGroup> allResourceGroups = resourceGroupService.listAll(resourceGroupFilterDTO, pageRequest);
+    List<ResourceGroup> permittedResourceGroups =
+        resourceGroupService.getPermittedResourceGroups(allResourceGroups.getContent());
+    permittedResourceGroups.sort(Comparator.comparing(ResourceGroup::getCreatedAt).reversed());
+
+    Page<ResourceGroup> resourceGroupPage =
+        PageUtils.getPage(permittedResourceGroups, pageRequest.getPageIndex(), pageRequest.getPageSize());
```

We should throw an exception of Missing permission in case permittedResourceGroups is empty.

---

Done